src/identity/components/GlobalHeader/GlobalHeaderDropdown/index.tsx

```diff
@@ -0,0 +1,201 @@
+    }
+  }
+
+  componentDidUpdate() {
+    // if the selected item changes, update the state
+    if (this.props.typeAheadSelectedOption !== this.state.selectedItem) {
+      this.setState({
+        selectedItem: this.props.typeAheadSelectedOption,
+      })
+    }
+  }
+
+  private dropdownButton = (
+    active: boolean,
+    onClick: (e: MouseEvent<HTMLElement>) => void
+  ) => {
+    const {
+      defaultButtonText,
+      dropdownButtonSize,
+      dropdownButtonIcon,
+    } = this.props
+    const {selectedItem} = this.state
+    return (
+      <Dropdown.Button
+        active={active}
+        onClick={onClick}
+        size={dropdownButtonSize}
+        icon={dropdownButtonIcon}
+        className="global-header--dropdown-button"
+      >
+        {selectedItem?.name || defaultButtonText}
+      </Dropdown.Button>
+    )
+  }
+
+  private toggleShowTypeAheadMenu = () => {
+    const {showTypeAheadMenu} = this.state
+    this.setState({showTypeAheadMenu: !showTypeAheadMenu})
+  }
+
+  private renderMainMenuOptions = () => {
+    const {mainMenuOptions} = this.props
+    return (
+      <div>
+        <hr className="line-break" />
+        {mainMenuOptions.map(value => {
+          const iconEl = <Icon glyph={value.iconFont} className="button-icon" />
+          const textEl = <span>{value.name}</span>
+          return (
+            <Dropdown.HrefItem key={value.name} href={value.href}>
+              {iconEl}
+              {textEl}
+            </Dropdown.HrefItem>
+          )
+        })}
+      </div>
+    )
+  }
+
+  private renderTypeAheadMenu = () => {
+    const {typeAheadMenuOptions} = this.props
+    const {selectedItem} = this.state
+    const {
+      typeAheadInputPlaceholder,
+      typeAheadOnSelectOption,
+      dropdownMenuStyle,
+    } = this.props
+    return (
+      <GlobalHeaderTypeAheadMenu
+        typeAheadPlaceHolder={typeAheadInputPlaceholder}
+        typeAheadMenuOptions={typeAheadMenuOptions}
+        onSelectOption={typeAheadOnSelectOption}
+        style={dropdownMenuStyle}
+        defaultSelectedItem={selectedItem}
+      />
+    )
+  }
+
+  private renderMenu = () => {
+    const {
+      mainMenuHeaderText,
+      dropdownMenuTheme = DropdownMenuTheme.None,
+      dropdownMenuStyle,
+      typeAheadMenuOptions,
+    } = this.props
+    const {showTypeAheadMenu} = this.state
+
+    const textEl = <span>{mainMenuHeaderText}</span>
+    const iconEl = (
+      <Icon
+        glyph={showTypeAheadMenu ? IconFont.CaretLeft_New : IconFont.Switch_New}
+        className="button-icon"
+      />
+    )
+    return (
+      <Dropdown.Menu
+        // testID={menuTestID}
+        theme={dropdownMenuTheme}
+        style={dropdownMenuStyle}
+      >
+        {/* Multi-org UI tickets #4051 and #4047, when user only has 1 account or 1 org, switch button is disabled */}
+        {typeAheadMenuOptions.length > 1 && (
+          <Dropdown.Item onClick={() => this.toggleShowTypeAheadMenu()}>
```

Comment:
We could move this anonymous function into a function defined outside of the render (or in this case return) to avoid unnecessary re-renders: [anonymous functions in react component attributes](https://stackoverflow.com/questions/42280526/how-unperformant-are-anonymous-functions-in-react-component-attributes)

Reply:
✅